src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlStreamNodeWriter.cs

```diff
@@ -384,38 +392,51 @@ protected unsafe int UnsafeGetUTF8Chars(char* chars, int charCount, byte[] buffe
                 fixed (byte* _bytes = &buffer[offset])
                 {
                     byte* bytes = _bytes;
-                    byte* bytesMax = &bytes[buffer.Length - offset];
                     char* charsMax = &chars[charCount];
 
-                    while (true)
+                    if (Sse41.IsSupported && charCount >= Vector128<short>.Count)
```

Comment:
This needs an optimized path for Arm64 too. You would get that for free if you just called Encoding.UTF8.

Reply:
If count is >16 it will now fall through to 'NonAscii' and call System.Encoding.GetBytes directly in order to get the optimized path

It is this cutoff I mentioned in "4." in the summary under "Feedback wanted ..." I am open to changing it to any other value and don't want to make a to large regression by choosing a to low value.

Reply:
I did belive arm64 supported SVE and 256bit vectors that is why I also included the generic Vector256 measurements as ["SimdVector256"](https://github.com/Daniel-Svensson/ClrExperiments/blob/97e6caa6a77c8c64f9d231b2647376e74043efe1/BinaryXmlBenchmarks/ConsoleApp1/Utf8Benchmarks.cs#L692)  in Benchmark results as a possible  alternative.

since i have no idea of how the unaligned loads/stores affect performance on arm i feel it is safer to fallback to System.encoding on arm than to write using generic Vector128 for now. (And also a bit by "generic" version having worse performance than direct sse,/avx usage)

Reply:
removed the code for now